python/cudf_polars/cudf_polars/dsl/ir.py

```diff
@@ -307,6 +307,22 @@ def _parquet_physical_types(
     return dict(zip(schema.keys(), [c.type() for c in df.tbl.columns()], strict=True))
 
 
+def _cast_literal_to_decimal(
+    side: expr.Expr, lit: expr.Literal, phys_type_map: dict[str, plc.DataType]
+) -> expr.Expr:
+    if isinstance(side, expr.Cast):
+        col = side.children[0]
+        assert isinstance(col, expr.Col)
+        name = col.name
+    else:
+        assert isinstance(side, expr.Col)
+        name = side.name
+    if phys_type_map[name].id() in _DECIMAL_IDS:
+        scale = abs(phys_type_map[name].scale())
```

Comment:
Minor nit, you could use a walrus to save a dict lookup here:
```suggestion
    if (type_ := phys_type_map[name]).id() in _DECIMAL_IDS:
        scale = abs(type_.scale())
```